lib/read.js

```diff
@@ -36,67 +43,56 @@ var Read = Juttle.proc.source.extend({
             this.baseQuery = this.addFilters(params.filter_ast);
         }
 
-        this.handleTimeOptions(options);
         this.addOptimizations(params.optimization_info);
 
         this.baseQuery = this.baseQuery.from(options.table).limit(this.getNextLimit());
 
-        //cannot order by time with any optimized reduce
-        if (this.timeField && !(params.optimization_info && params.optimization_info.type === 'reduce')) {
-            this.baseQuery = this.baseQuery.orderBy(this.timeField || 'time', this.tailOptimized ? 'desc' : 'asc');
+        //cannot order by time if no reduce optimization
+        if (this.timeField && this.optimization_info.type !== 'reduce') {
+            this.baseQuery = this.baseQuery.orderBy(this.timeField, this.tailOptimized ? 'desc' : 'asc');
         }
-    },
+    }
+
+    periodicLiveRead() {
+        return !!this.timeField;
+    }
 
-    validateOptions: function(options) {
-        var unknown = _.difference(_.keys(options), this.allowed_options);
+    validateOptions(options) {
+        var unknown = _.difference(_.keys(options), ALLOWED_OPTIONS);
         if (unknown.length > 0) {
-            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+            throw new errors.runtimeError('RT-UNKNOWN-OPTION-ERROR', {
                 proc: 'read-sql',
                 option: unknown[0]
             });
         }
-
         if (_.has(options, 'table') === _.has(options, 'raw')) {
-            throw this.compile_error('RT-MISSING-OPTION-ERROR', { option: "choose only one: table,raw" });
+            throw new Error("choose only one option: table,raw.");
```

Comment:
Did you switch to just throwing an error because the message for RT-MISSING-OPTION-ERROR was not appropriate? If so you could try using RT-INCOMPATIBLE-OPTION-ERROR which has a {option} and {other}.